OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -236,20 +236,18 @@ public override bool HandleKeyPress(KeyInput e)
 				}
 				else if (key == Game.Settings.Keys.SelectUnitsByTypeKey)
 				{
-					var selectedTypes = World.Selection.Actors
+					var selectedClasses = World.Selection.Actors
 						.Where(x => x.Owner == player)
-						.Select(a => a.Info);
+						.Select(a => a.Trait<Selectable>().Class);
 
-					Func<Actor, bool> cond = a => a.Owner == player && selectedTypes.Contains(a.Info);
-					var tl = worldRenderer.Viewport.TopLeft;
-					var br = worldRenderer.Viewport.BottomRight;
-					var newSelection = SelectActorsInBox(World, tl, br, cond);
+					Func<Actor, bool> cond = a => a.Owner == player && selectedClasses.Contains(a.Trait<Selectable>().Class);
+					var newSelection = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight, cond);
 
-					if (newSelection.Count() > selectedTypes.Count())
+					if (newSelection.Count() > selectedClasses.Count())
```

Comment:
`selectedClasses` contains one `string` object per selected unit before invoking select-by-key.
`newSelection` contains one `Actor` object per selected unit after all actors on the screen with the same selection class have been selected.

`selectedClasses.Count()` gives exactly the number of units selected before selecting everyone on the screen
`newSelection.Count()` gives exactly the number of units selected after selecting everyone on the screen